src/components/ec-donut/ec-donut.spec.js

```diff
@@ -0,0 +1,72 @@
+import { mount } from '@vue/test-utils';
+import EcDonut from './ec-donut.vue';
+import { withMockedConsole } from '../../../tests/utils/console';
+
+describe('EcDonut', () => {
+  function mountDonut(props, mountOpts) {
+    return mount(EcDonut, {
+      propsData: {
+        used: 20,
+        amount: 100,
+        ...props,
+      },
+      ...mountOpts,
+    });
+  }
+
+  it('should throw if no props were given', () => {
+    withMockedConsole((errorSpy) => {
+      mount(EcDonut);
+      expect(errorSpy).toHaveBeenCalledTimes(2);
+      expect(errorSpy.mock.calls[0][0]).toContain('Missing required prop: "amount"');
+      expect(errorSpy.mock.calls[1][0]).toContain('Missing required prop: "used"');
+    });
+  });
+
+  it('should display only with a the used and amount are given', () => {
+    const wrapper = mountDonut();
+    expect(wrapper.element).toMatchSnapshot();
+  });
+
+  it('should display with the half used', () => {
+    const wrapper = mountDonut({ used: 50, amount: 100 });
+    expect(wrapper.element).toMatchSnapshot();
+  });
+
+  it('should display with the empty used', () => {
+    const wrapper = mountDonut({ used: 0 });
+    expect(wrapper.element).toMatchSnapshot();
+  });
+
+  it('should display with the full used', () => {
+    const wrapper = mountDonut({ used: 100, amount: 100 });
```

Comment:
wtf, marked as resolved still not added the negative spec?

Reply:
Done now!